### PR TITLE
NameValueList - Allow user to define different gap depending on pairProp direction

### PR DESCRIPTION
**Pull request by taysea**
#### What does this PR do?
Adds `theme.nameValueList.pair.column.gap` to theme to allow custom specification of gap for when a pair is in a column.

#### Where should the reviewer start?
src/js/components/NameValueList/NameValueList.js

#### What testing has been done on this PR?
Tested in storybook examples, updated snapshots and confirmed changes are expected.

#### How should this be manually tested?
Storybook, inspect. OR update storybook example locally.

#### Screenshots (if appropriate)
Row/Desktop:
Screen Shot 2021-11-05 at 9 49 20 AM

Column/Mobile:
Screen Shot 2021-11-05 at 9 49 31 AM

#### Do the grommet docs need to be updated?
Yes.

#### Should this PR be mentioned in the release notes?
No. Component hasn't been released.

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.

### CODE REVIEW

File: src/js/components/NameValueList/stories/typescript/Themed.tsx [hunk 1]
```diff
@@ -7,6 +7,12 @@ import { data } from '../data';
 // Type annotations can only be used in TypeScript files.
 // Remove ': ThemeType' if you are not using Typescript.
 const customTheme: ThemeType = deepMerge(grommet, {
+  nameValueList: {
+    gap: {
+      column: 'xlarge',
+      row: 'xlarge',
```
Comment (britt6612) [inline src/js/components/NameValueList/stories/typescript/Themed.tsx:13]: 👍

**Review by britt6612 — APPROVED** [review]

**Comment by britt6612** [comment]
well just need to add to the site if the naming is what we decide on I can do this since Im adding the `margin`

File: src/js/themes/base.d.ts [hunk 2]
```diff
@@ -1151,7 +1151,16 @@ export interface ThemeType {
     };
   };
   nameValueList?: {
-    gap?: { row?: GapType; column?: GapType };
+    gap?: { 
+      pairDirectionColumn?: {
```
Comment (ericsoderberghp) [inline src/js/themes/base.d.ts:1155]: This doesn't seem consistent with our pattern elsewhere of keying off of state at the root of the component, for example `theme.button.active`. Did you consider something like `theme.nameValueList.column.gap` and `theme.nameValueList.grid.gap`?

Comment (halocline) [reply src/js/themes/base.d.ts:1155]: I also like the idea of keeping the state at the root, then setting attributes specifically for each particular state.

**Review by ericsoderberghp — APPROVED** [review]

**Review by britt6612 — APPROVED** [review]

**Review by kevingill55 — COMMENTED** [review]
This looks good on my end! One comment about whether or not we should add the `Themed` storybook example to a Custom Themed directory (to adhere to our new storybook theming conventions)

**Comment by britt6612** [comment]
> This looks good on my end! One comment about whether or not we should add the `Themed` storybook example to a Custom Themed directory (to adhere to our new storybook theming conventions)

good point pushing this change up now

**Pull request merged**